File: backend/src/main/java/com/recette/allenchang/backend/recipes/dto/responses/RecipeResponse.java
package com.recette.allenchang.backend.recipes.dto.responses;

import java.util.List;
import java.util.UUID;

public record RecipeResponse(
                UUID id,
                String title,
                String description,
                String imageurl,
                List<IngredientResponse> ingredients,
                List<String> steps,
                List<String> tags,
                String difficulty,
                Integer servingSize,
                Integer cookTimeInMinutes) {
}

File: frontend-ios/DTOs/Responses/RecipeResponse.swift
import Foundation

struct RecipeResponse: Codable {
    let id: String
    let title: String
    let description: String
    let imageurl: String?
    let ingredients: [IngredientDTO]
    let steps: [String]
    let tags: [String]
    let difficulty: String?
    let servingSize: Int?
    let cookTimeInMinutes: Int?
    let isPublic: Bool?
    let owner: UserResponse?
    let isSavedByCurrentUser: Bool?
}